src/pages/instances/MigrateInstanceModal.tsx

```diff
@@ -1,197 +1,100 @@
-import { FC, KeyboardEvent, useState } from "react";
-import {
-  ActionButton,
-  Button,
-  MainTable,
-  Modal,
-} from "@canonical/react-components";
+import { FC, KeyboardEvent, ReactNode, useState } from "react";
+import { Modal } from "@canonical/react-components";
 import { LxdInstance } from "types/instance";
-import { useQuery } from "@tanstack/react-query";
-import { queryKeys } from "util/queryKeys";
-import { fetchClusterMembers } from "api/cluster";
-import Loader from "components/Loader";
-import ScrollableTable from "components/ScrollableTable";
+import { useSettings } from "context/useSettings";
+import { isClusteredServer } from "util/settings";
+import FormLink from "components/FormLink";
+import ClusterMemberMigrationTable from "./ClusterMemberMigrationTable";
+import BackLink from "components/BackLink";
+import StoragePoolMigrationTable from "./StoragePoolMigrationTable";
 
 interface Props {
   close: () => void;
-  migrate: (target: string, instance: LxdInstance) => void;
   instance: LxdInstance;
 }
 
-const MigrateInstanceModal: FC<Props> = ({ close, migrate, instance }) => {
-  const { data: members = [], isLoading } = useQuery({
-    queryKey: [queryKeys.cluster, queryKeys.members],
-    queryFn: fetchClusterMembers,
-  });
+type MigrationType = "cluster-member" | "storage-pool";
 
-  const [selectedMember, setSelectedMember] = useState("");
+const MigrateInstanceModal: FC<Props> = ({ close, instance }) => {
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const [type, setType] = useState<MigrationType | null>(null);
```

Comment:
```suggestion
  const [type, setType] = useState<MigrationType | null>(isClustered ? null : "root storage pool");
```
Maybe this makes below simpler, we can hide the back link and remove the other special cases?

Reply:
The simplest I can reduce the code below is something like 
```
const modalTitle = !type ? (
    "Choose migration method"
  ) : (
    <>
      {isClustered && (
        <BackLink
          title={target ? "Confirm migration" : `Choose new ${type}`}
          onClick={handleGoBack}
          linkText={target ? `Choose ${type}` : "Choose migration method"}
        />
      )}
      {!isClustered && (target ? "Confirm migration" : `Choose new ${type}`)}
    </>
  );
```
Is this what you had in mind? it's less code but I think it's still a little messy